Src/Newtonsoft.Json/Serialization/NamingStrategy.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace Newtonsoft.Json.Serialization
+{
+    /// <summary>
+    /// A base class for resolving how property names and dictionary keys are serialized.
+    /// </summary>
+    public abstract class NamingStrategy
+    {
+        /// <summary>
+        /// A flag indicating whether dictionary keys should be processed.
+        /// Defaults to <c>false</c>.
+        /// </summary>
+        public bool ProcessDictionaryKeys { get; set; }
+
+        /// <summary>
+        /// A flag indicating whether explicitly specified property names,
+        /// e.g. a property name customized with a <see cref="JsonPropertyAttribute"/>, should be processed.
+        /// Defaults to <c>false</c>.
```

Comment:
This seems like a sensible default. Would you think it is worth having tests for the default behaviour? If this were to change by accident it would be a breaking change